x-pack/plugins/actions/server/lib/action_task_params_utils.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectAttribute, SavedObjectReference } from 'src/core/server/types';
+import { RelatedSavedObjects } from './related_saved_objects';
+
+export const ACTION_REF_NAME = `actionRef`;
+
+export function extractSavedObjectReferences(
+  actionId: string,
+  isPreconfigured: boolean,
+  relatedSavedObjects?: RelatedSavedObjects
+): {
+  references: SavedObjectReference[];
+  relatedSavedObjectWithRefs?: RelatedSavedObjects;
+} {
+  const references: SavedObjectReference[] = [];
+  const relatedSavedObjectWithRefs: RelatedSavedObjects = [];
+
+  // Add action saved object to reference if it is not preconfigured
+  if (!isPreconfigured) {
+    references.push({
+      id: actionId,
+      name: ACTION_REF_NAME,
+      type: 'action',
+    });
+  }
+
+  // Add related saved objects, if any
+  (relatedSavedObjects ?? []).forEach((relatedSavedObject, index) => {
+    relatedSavedObjectWithRefs.push({
+      ...relatedSavedObject,
+      id: `related_${relatedSavedObject.type}_${index}`,
+    });
+    references.push({
+      id: relatedSavedObject.id,
+      name: `related_${relatedSavedObject.type}_${index}`,
+      type: relatedSavedObject.type,
+    });
+  });
+
+  return {
+    references,
+    ...(relatedSavedObjects ? { relatedSavedObjectWithRefs } : {}),
+  };
+}
+
+export function injectSavedObjectReferences(
+  references: SavedObjectReference[],
+  relatedSavedObjects?: RelatedSavedObjects
+): { actionId?: string; relatedSavedObjects?: SavedObjectAttribute } {
+  references = references ?? [];
+
+  // Look for for the action id
+  const action = references.find((ref) => ref.name === ACTION_REF_NAME);
+
+  const injectedRelatedSavedObjects = (relatedSavedObjects ?? []).flatMap((relatedSavedObject) => {
+    const reference = references.find((ref) => ref.name === relatedSavedObject.id);
+
+    // These are used to provide context in the event log so we will not throw an error
```

Comment:
I'm not sure I understand this comment. Can you reword or perhaps provide code links to the affected areas?

Reply:
I tried to clarify the comment in this commit: https://github.com/elastic/kibana/pull/108964/commits/105e57e4a9cff11ab9bbb34aae6831bb0530d1b3

Reply:
Thanks!